cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -331,7 +323,13 @@ func getSchedulerDeployment(cfg *kubeadmapi.MasterConfiguration) ext.Deployment
 			},
 		},
 	}
-	return d
+}
+
+func prependSelfHostedFlock(name string, args []string) []string {
+	// TODO: flock has to be inside the container, and a shell must exist with this approach
+	// Since the flock binary is present at /bin/flock in a busybox image and /usr/bin/flock in a debian image, we can't hardcode the path
```

Comment:
OK it now seems we're moving to scratch containers with just the binaries that are supposed to run so I now agree with @luxas that we will need to bind-mount the host flock binary to the containers. I'm just worried about the host-path for different distros.